pages/activity/index.js

```diff
@@ -71,10 +90,12 @@ export default function HomePage({
       )}
 
       <ActivityList
-        activities={activities}
+        activities={filteredActivities != "" ? filteredActivities : listedActivities}
```

Comment:
this can be replaced with listedActivities, since depending on the searchTerm the random or filtered Activities are shown.